### PR TITLE
MULE-16353: File connectors fails to fail when it fails to lock a file

### PR DESCRIPTION
**Pull request by SebaElizalde**

### CODE REVIEW

File: src/test/java/org/mule/extension/ftp/FtpWriteTestCase.java [hunk 1]
```diff
@@ -199,4 +221,27 @@ private String doWriteOnExistingFile(FileWriteMode mode) throws Exception {
...
+    return (new InputStream() {
+
+      String text = "Hello World!";
+      char[] textArray = text.toCharArray();
+      int index = -1;
+
+      @Override
+      public int read() throws IOException {
+        try {
+          Thread.sleep(10);
```
Comment (sonar-mulesoft) [inline src/test/java/org/mule/extension/ftp/FtpWriteTestCase.java]: Code Smell Code Smell: Remove this use of "Thread.sleep()". ([squid:S2925](https://sonarqube.kbuild.msap.io/coding_rules#rule_key=squid%3AS2925))

[See it in SonarQube](https://sonarqube.kbuild.msap.io/project/issues?id=mule-core-extensions-mule-ftp-connector&pullRequest=101&issues=AWkMI-sJotIXnmsUtDdh&open=AWkMI-sJotIXnmsUtDdh)

[ISSUE_KEY]: AWkMI-sJotIXnmsUtDdh

File: src/test/java/org/mule/extension/ftp/FtpWriteTestCase.java [hunk 2]
```diff
@@ -139,6 +142,26 @@ public void writeStaticContent() throws Exception {
...
+    doWrite("writeWithLock", path, HELLO_WORLD, CREATE_NEW, false);
+
+    String content = toString(readPath(path).getPayload().getValue());
+    assertThat(content, is(HELLO_WORLD));
+  }
+
+  //TODO: MULE-16515 ignore this test until issue is fixed.
+  @Test
+  @Ignore
+  public void writeOnLockedFile() throws Exception {
```
Comment (sonar-mulesoft) [inline src/test/java/org/mule/extension/ftp/FtpWriteTestCase.java]: Code Smell Code Smell: Fix or remove this skipped unit test ([squid:S1607](https://sonarqube.kbuild.msap.io/coding_rules#rule_key=squid%3AS1607))

[See it in SonarQube](https://sonarqube.kbuild.msap.io/project/issues?id=mule-core-extensions-mule-ftp-connector&pullRequest=101&issues=AWkMI-sJotIXnmsUtDdg&open=AWkMI-sJotIXnmsUtDdg)

[ISSUE_KEY]: AWkMI-sJotIXnmsUtDdg

File: pom.xml [hunk 3]
```diff
@@ -91,6 +98,14 @@
             <classifier>mule-plugin</classifier>
             <scope>test</scope>
         </dependency>
+
+        <dependency>
+            <groupId>org.mule.module</groupId>
+            <artifactId>mule-java-module</artifactId>
+            <version>1.3.0-SNAPSHOT</version>
```
Comment (ndinu) [inline pom.xml]: Use released version if possible

File: src/test/java/org/mule/extension/ftp/FtpWriteTestCase.java (hunk 2 above)
Comment (sonar-mulesoft) [inline src/test/java/org/mule/extension/ftp/FtpWriteTestCase.java]: Code Smell Code Smell: Fix or remove this skipped unit test ([squid:S1607](https://sonarqube.kbuild.msap.io/coding_rules#rule_key=squid%3AS1607))

[See it in SonarQube](https://sonarqube.kbuild.msap.io/project/issues?id=mule-core-extensions-mule-ftp-connector&pullRequest=101&issues=AWkMI-sJotIXnmsUtDdg&open=AWkMI-sJotIXnmsUtDdg)

[ISSUE_KEY]: AWkMI-sJotIXnmsUtDdg

File: src/test/java/org/mule/extension/ftp/FtpWriteTestCase.java (hunk 1 above)
Comment (sonar-mulesoft) [inline src/test/java/org/mule/extension/ftp/FtpWriteTestCase.java]: Code Smell Code Smell: Remove this use of "Thread.sleep()". ([squid:S2925](https://sonarqube.kbuild.msap.io/coding_rules#rule_key=squid%3AS2925))

[See it in SonarQube](https://sonarqube.kbuild.msap.io/project/issues?id=mule-core-extensions-mule-ftp-connector&pullRequest=101&issues=AWkMI-sJotIXnmsUtDdh&open=AWkMI-sJotIXnmsUtDdh)

[ISSUE_KEY]: AWkMI-sJotIXnmsUtDdh

File: src/test/java/org/mule/extension/ftp/FtpWriteTestCase.java [hunk 4]
```diff
@@ -199,4 +231,27 @@ private String doWriteOnExistingFile(FileWriteMode mode) throws Exception {
...
+    return (new InputStream() {
+
+      String text = "Hello World!";
+      char[] textArray = text.toCharArray();
+      int index = -1;
+
+      @Override
+      public int read() throws IOException {
+        try {
+          Thread.sleep(10);
```
Comment (sonar-mulesoft) [inline src/test/java/org/mule/extension/ftp/FtpWriteTestCase.java]: Code Smell Code Smell: Remove this use of "Thread.sleep()". ([squid:S2925](https://sonarqube.kbuild.msap.io/coding_rules#rule_key=squid%3AS2925))

[See it in SonarQube](https://sonarqube.kbuild.msap.io/project/issues?id=mule-core-extensions-mule-ftp-connector&pullRequest=101&issues=AWkMI-sJotIXnmsUtDdh&open=AWkMI-sJotIXnmsUtDdh)

[ISSUE_KEY]: AWkMI-sJotIXnmsUtDdh

File: src/test/java/org/mule/extension/ftp/FtpWriteTestCase.java [hunk 5]
```diff
@@ -139,6 +144,34 @@ public void writeStaticContent() throws Exception {
...
+    doWrite("writeWithLock", path, HELLO_WORLD, CREATE_NEW, false);
+
+    String content = toString(readPath(path).getPayload().getValue());
+    assertThat(content, is(HELLO_WORLD));
+  }
+
+  //TODO: MULE-16515 ignore this test until issue is fixed.
+  @Test
+  @Ignore
+  public void writeOnLockedFile() throws Exception {
```
Comment (sonar-mulesoft) [inline src/test/java/org/mule/extension/ftp/FtpWriteTestCase.java]: Code Smell Code Smell: Fix or remove this skipped unit test ([squid:S1607](https://sonarqube.kbuild.msap.io/coding_rules#rule_key=squid%3AS1607))

[See it in SonarQube](https://sonarqube.kbuild.msap.io/project/issues?id=mule-core-extensions-mule-ftp-connector&pullRequest=101&issues=AWkMI-sJotIXnmsUtDdg&open=AWkMI-sJotIXnmsUtDdg)

[ISSUE_KEY]: AWkMI-sJotIXnmsUtDdg

**Pull request merged**